mlos_bench/mlos_bench/tests/environments/composite_env_test.py

```diff
@@ -54,6 +55,47 @@ def composite_env(tunable_groups: TunableGroups) -> CompositeEnv:
                         "range": [60, 120],
                         "metrics": ["score"],
                     }
+                },
+                {
+                    "name": "Composite Child 3",
+                    "class": "mlos_bench.environments.composite_env.CompositeEnv",
+                    "config": {
+                        "const_args": {
+                            # This should be propagated to the children, but not up to the parent.
+                            "vmName": "Nested Mock VM",
+                            "EnvId": 3
+                        },
+                        "required_args": ["vmName"],
+                        "children": [
+                            {
+                                "name": "Nested Mock Environment 1",
+                                "class": "mlos_bench.environments.mock_env.MockEnv",
+                                "config": {
+                                    "tunable_params": ["provision"],
+                                    "const_args": {
+                                        "vmName": "Placeholder VM",
+                                    },
+                                    "required_args": ["vmName", "someConst"],
+                                    "range": [60, 120],
+                                    "metrics": ["score"],
+                                }
+                            },
+                            {
+                                "name": "Nested Mock Environment 2",
+                                "class": "mlos_bench.environments.mock_env.MockEnv",
+                                "config": {
+                                    "tunable_params": ["boot"],
+                                    "const_args": {
+                                        "vmName": "Placeholder VM",
+                                        "someConst": "leaf"
+                                    },
+                                    "required_args": ["vmName", "someConst"],
+                                    "range": [60, 120],
+                                    "metrics": ["score"],
```

Comment:
@motus 
Another issue is some of the repetition in these configs.  I wonder if we could have a sort "inheritable" args vs. not config.
Thoughts?

Reply:
yeah, I was thinking about it, but I've decided to avoid any big changes to the framework for now and focus on the experiments; later we can think of an elegant way of solving this issue - but maybe we don't even have to fix it in JSON but instead introduce some high-level DSL that would compile into JSONs and scripts that we use now. Again, let's discuss when we have time